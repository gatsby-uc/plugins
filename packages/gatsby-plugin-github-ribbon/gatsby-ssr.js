const gatsbySsr = require("./dist/gatsby-ssr");

module.exports = gatsbySsr;
