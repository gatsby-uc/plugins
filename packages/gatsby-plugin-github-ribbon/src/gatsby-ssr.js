import React from "react"
import { withPrefix } from "gatsby"

exports.onRenderBody = ({ setPreBodyComponents }, pluginOptions) => {
  const { project, position } = pluginOptions

  var baseImgStyle = {
    position: `absolute`,
    top: 0,
    border: 0,
  }
  var imgStyle

  if (position === `right`) {
    imgStyle = {
      ...baseImgStyle,
      right: 0,
    }
  } else {
    imgStyle = {
      ...baseImgStyle,
      left: 0,
    }
  }

  setPreBodyComponents([
    <header key={`gatsby-plugin-github-ribbon-header`}>
      <a href={project} rel={`noopener`} target={`_blank`}>
        <img
          style={imgStyle}
          src={withPrefix(`/github_ribbon.png`)}
          alt="Fork me on GitHub"
        ></img>
      </a>
    </header>,
  ])
}
