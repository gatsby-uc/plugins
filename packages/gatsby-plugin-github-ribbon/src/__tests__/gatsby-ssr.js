import { onRenderBody } from "../gatsby-ssr";

describe(`adds github ribbon to html`, () => {
  it.each`
    position
    ${`left`}
    ${`right`}
  `(`create html for injecting if config is set - $position`, ({ position }) => {
    //setup test data
    const pluginOptions = {
      project: `https://github.com/gatsbyjs/gatsby`,
      color: `red`,
      position: position,
    };

    const setPreBodyComponents = jest.fn();

    //fake test
    onRenderBody(
      {
        setPreBodyComponents,
      },
      pluginOptions,
    );

    expect(setPreBodyComponents).toMatchSnapshot();
    expect(setPreBodyComponents).toHaveBeenCalledTimes(1);
  });
});
