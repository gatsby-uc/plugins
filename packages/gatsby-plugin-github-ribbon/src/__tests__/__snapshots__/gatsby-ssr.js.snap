// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`adds github ribbon to html create html for injecting if config is set - left 1`] = `
[MockFunction] {
  "calls": [
    [
      [
        <header>
          <a
            href="https://github.com/gatsbyjs/gatsby"
            rel="noopener"
            target="_blank"
          >
            <img
              alt="Fork me on GitHub"
              src="/github_ribbon.png"
              style={
                {
                  "border": 0,
                  "left": 0,
                  "position": "absolute",
                  "top": 0,
                }
              }
            />
          </a>
        </header>,
      ],
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`adds github ribbon to html create html for injecting if config is set - right 1`] = `
[MockFunction] {
  "calls": [
    [
      [
        <header>
          <a
            href="https://github.com/gatsbyjs/gatsby"
            rel="noopener"
            target="_blank"
          >
            <img
              alt="Fork me on GitHub"
              src="/github_ribbon.png"
              style={
                {
                  "border": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            />
          </a>
        </header>,
      ],
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
